sphinx-docs/screenshot.py

```diff
@@ -0,0 +1,328 @@
+    subprocess.wait()
+    if display:
+        display.stop()
+
+def _parse_focus(arg_str):
+    """ Returns id and annotation after splitting input string.
+
+    First argument should be the jQuery-style selector. An optional 
+    annotation can follow if separated by a separator '|'. Initial
+    whitespace after the '|' will be ignored.
+
+    Example inputs:
+        #an_id
+        #another_id | With an annotation
+        form.foo input.radio | The quick brown fox 
+    """
+    split_str = arg_str.split('|', 1)
+    if len(split_str) == 1:
+        return {'id': split_str[0].rstrip(), 'annotation': ''}
+    else:
+        return {'id': split_str[0].rstrip(), 'annotation': split_str[1].lstrip()}
+
+def _parse_command(command):
+    """" Parses a command into action and options.
+
+    Returns a dictionary with following keys:
+       selector (string): the selector to identify the element
+       action (string): the action type (if it's recognized)
+       options (list): a list of options
+
+    Raises an error if action type is not recognized or if options are invalid.
+
+    Example inputs:
+        #sync_button click
+        NEXT send_keys some keys
+        NEXT send_keys special characters like TAB and ENTER can be used like this
+
+    Note that 'TAB', 'ENTER', and 'BACKSPACE' all have special meaning for send_keys
+    """
+    command_args = command.split()
+    if not command_args:
+        return None
+    else:
+        selector = command_args[0]
+        action = command_args[1]
+        options = command_args[2:]
+
+    if action not in ('click', 'send_keys', 'submit', ''):
+        raise ActionError(action)
+
+    # Validate input options for the specified action.
+    if action == 'click' or action == 'submit':
+        if options:
+            raise OptionError("The action '%s' must not contain any arguments whereas supplied arguments: '%s'." % (action, repr(options)))
+
+    return {'selector': selector, 'action': action, 'options': options}
+
+def _parse_login(username, password, submit=""):
+    """" Parses a LOGIN command.
+
+    Returns a dictionary with following keys:
+       runhandler (string):  "_login_handler".
+       args (dict) : A dictionary of arguments with following keys:
+         username (string):  the username.
+         password (string): password.
+         submit (bool): True if login form is to be submitted, false otherwise.
+    """
+    submit_bool = True if submit == "submit" else False
+    args = {'username': username, 'password': password, 'submit': submit_bool}
+    return {'runhandler': '_login_handler', 'args': args}
+
+def _parse_nav_steps(arg_str):
+    """ Here's how to specify the navigation steps:
+
+        1. selector action [options] ["|" selector action [options]] ...
+        2. aliased_action_sequence [options]
+
+        selector will identify the element... for now we'll just implement
+            selection by id e.g. "#username-field"
```

Comment:
Sample args here would also be good. Not sure if this is enough for people to understand what's going on in this function.